hack/make-rules/test.sh

```diff
@@ -225,9 +225,10 @@ produceJUnitXMLReport() {
   junit_xml_filename="${junit_filename_prefix}.xml"
 
   if ! command -v gotestsum >/dev/null 2>&1; then
-    kube::log::error "gotestsum not found; please cd to hack/tools and install with " \
-      "GO111MODULE=on go install gotest.tools/gotestsum"
-    return
+    kube::log::status "gotestsum not found; installing from hack/tools"
+    pushd "${KUBE_ROOT}/hack/tools" >/dev/null
+      GO111MODULE=on go install gotest.tools/gotestsum
```

Comment:
I don't see us setting up a GOBIN / adding it to path anywhere (am I forgetting this happening in one of the sourced files?)

We want to avoid clobbering the system installed copy and usually have a GOBIN under _output like:
https://github.com/kubernetes/kubernetes/blob/792b4b1e32394a13c27741f71f14dd134d663a5f/hack/verify-govet-levee.sh#L30-L31

Reply:
> We want to avoid clobbering the system installed copy

this only runs if we can't find `gotestsum` in the path already

kube::golang::setup_env sets GOPATH and clears GOBIN, so go install goes under `_output/local/go/bin`

Reply:
we can probably dedupe this in other files then, but not in this PR 🙃